tests/gpts/github/test-config/github_advanced_actions.ts

```diff
@@ -0,0 +1,256 @@
+        }, 120000);
+
+        /**
+         * Verification to check if Red Hat Developer Hub created the gitops repository with wrkflow file
+         */
+        it(`verifies if component ${gptTemplate} have a valid gitops repository and there exists a workflow file`, async () => {
+            expect(await gitHubClient.checkIfRepositoryExists(githubOrganization, `${repositoryName}-gitops`)).toBe(true);
+            expect(await gitHubClient.checkIfFolderExistsInRepository(githubOrganization, repositoryName, '.github/workflows/build-and-update-gitops.yml')).toBe(true);
+        }, 120000);
+
+        /**
+         * Creates secrets for GitHub Actions Workflow
+         */
+        it (`creates env variables in repo`, async () => {
+            const repoDict = [
+                {
+                    repoName: repositoryName,
+                    workflowPath: '.github/workflows/build-and-update-gitops.yml'
+                },
+                {
+                    repoName: `${repositoryName}-gitops`,
+                    workflowPath: '.github/workflows/gitops-promotion.yml'
+                }
+            ];
+            for (const repoData of repoDict) {
+                await setGitHubActionSecrets(gitHubClient, kubeClient, githubOrganization, repoData.repoName, imageRegistry, RHTAPRootNamespace);
+                expect(await gitHubClient.updateWorkflowFileToEnableSecrets(githubOrganization, repoData.repoName, repoData.workflowPath)).not.toBe(undefined);
+            }
+
+        }, 600000);
+
+        /**
+         * Trigger and wait for Actions job to finish
+         */
+        it(`trigger and wait for ${gptTemplate} GitHub Actions job`, async () => {
+            expect(await gitHubClient.getLatestWorkflowRunStatus(githubOrganization, repositoryName, "TSSC-Build-Attest-Scan-Deploy")).toBe("success");
+        }, 300000);
+
+        /**
+         * Obtain the openshift Route for the component and verify that the previous builded image was synced in the cluster and deployed in development environment
+         */
+        it('container component is successfully synced by gitops in development environment', async () => {
+            console.log("syncing argocd application in development environment");
+            await checkComponentSyncedInArgoAndRouteIsWorking(kubeClient, backstageClient, developmentNamespace, developmentEnvironmentName, repositoryName, stringOnRoute);
+        }, 900000);
+
+        /**
+         * Trigger a promotion Pull Request in Gitops repository to promote development image to stage environment
+         */
+        it('trigger pull request promotion to promote from development to stage environment', async () => {
+            const getImage = await gitHubClient.extractImageFromContent(githubOrganization, `${repositoryName}-gitops`, repositoryName, developmentEnvironmentName);
+
+            if (getImage !== undefined) {
```

Comment:
can you please move this check in gitHubClient.extractImageFromContent() or update it there? Thank you!